tests/crypto/bigint_test.cpp

```diff
@@ -46,8 +46,6 @@ BOOST_AUTO_TEST_CASE(bigint_test_2)
         BOOST_CHECK( bi_accu >= a_1 );
     } while ( bi_accu.log2() <= 128 );
 
-    bi_accu = bi_accu;
-
```

Comment:
I checked that bigint does properly ignore a self-assignment (and self-move), and this line causes warnings. I think we can safely do away with it.

Reply:
IIRC I added this line on purpose, to test just that. Can you trick the compiler? Perhaps by assigning a reference to itself or sth?

Reply:
OK, so I've added it back but with enough nonsense to confuse the compiler into just doing it and not warning us about it. How's that?